plottr/plot/mpl.py

```diff
@@ -881,14 +921,11 @@ def _plotTypeFromToolBar(self, plotType: PlotType) -> None:
             self.plotType = plotType
             self._plotData(adjustSize=True)
 
-    @Slot(bool)
-    def _complexPreferenceFromToolBar(self, magPhasePreferred: bool) -> None:
-        if magPhasePreferred:
-            self.complexPreference = ComplexRepresentation.magAndPhase
-        else:
-            self.complexPreference = ComplexRepresentation.realAndImag
-
-        self._plotData(adjustSize=True)
+    @Slot(ComplexRepresentation)
+    def _complexPreferenceFromToolBar(self, complexRepresentation: ComplexRepresentation) -> None:
+        if complexRepresentation is not self.complexRepresentation:
+            self.complexRepresentation = complexRepresentation
+            self._plotData(adjustSize=True)
```

Comment:
Please revert back so that this method sets `self.complexPreference`

Reply:
I am going to remove complexPreference completely in the next commit as it is not going to be used anywhere. I am keeping this function as it is, and I fixed the default check. Please look at the next commit.